lib/m2ts/caption-stream.js

```diff
@@ -264,9 +370,56 @@ var createDisplayBuffer = function() {
   return result;
 };
 
-var Cea608Stream = function() {
+var Cea608Stream = function(track) {
   Cea608Stream.prototype.init.call(this);
 
+  this.track_ = track || 1;
+  switch (this.track_) {
+    case 1:
+      this.field_ = 0;
+      this.dataChannel_ = 0;
+      break;
+    case 2:
+      this.field_ = 0;
+      this.dataChannel_ = 1;
+      break;
+    case 3:
+      this.field_ = 1;
+      this.dataChannel_ = 0;
+      break;
+    case 4:
+      this.field_ = 1;
+      this.dataChannel_ = 1;
+      break;
+  }
+
+  // The following attributes have these uses:
+  // ext_ :    char0 for mid-row codes, and the base for extended
+  //           chars (ext_+0, ext_+1, and ext_+2 are char0s for
+  //           extended codes)
+  // control_: char0 for control codes, except byte-shifted to the
+  //           left so that we can do this.control_ | CONTROL_CODE
+  // offset_:  char0 for tab offset codes
+  //
+  // It's also worth noting that I've found control codes for field-2
+  // captions to be spec-breaking; they should be 0x14 or 0x1c, but
+  // come as 0x15 and 0x1d. That's why there's the "| field" on the
+  // control value.
```

Comment:
Woops! Thanks for catching this. I'd written this comment mostly as a note to myself so that I didn't forget to mention it in the PR. I later found where in the CEA608 spec it mentions this, but neglected to do update this comment with that information, which I will do now.